sematic/api/endpoints/runs.py

```diff
@@ -295,6 +295,25 @@ def get_logs_endpoint(user: Optional[User], run_id: str) -> flask.Response:
     )
 
     payload = dict(content=asdict(result))
+
+    # when none cursors is set, it means we are doing the initial pull
+    # and we can determine if we can continue in either direction
+    # this overrides the value from the `result`.
+    if (
+        kwargs["reverse_cursor_token"] is None
+        and kwargs["forward_cursor_token"] is None
+    ):
+        if kwargs["reverse"]:
+            run = get_run(run_id)
+
+            payload["content"]["can_continue_forward"] = run.future_state not in [
+                FutureState.RESOLVED.value,
+                FutureState.FAILED.value,
+                FutureState.NESTED_FAILED.value,
+            ]
+        else:
+            payload["content"]["can_continue_backward"] = False
+
```

Comment:
Will this logic also catch when the user has clicked "Load previous" enough time to reach the top?

Reply:
"Load previous" will attach a `forward_cursor_token` in the request URL, and will not go through this code. When `forward_cursor_token` is used, the code will behave correctly, in that, if there are no more logs before, it will emit `can_continue_backward: False` correctly.